packages/block-editor/src/store/actions.js

```diff
@@ -1111,6 +1111,24 @@ export const mergeBlocks =
 			return;
 		}
 
+		// If previous block is empty, remove previous block and update selection.
+		const blockAHasEmptyContent = blockA?.attributes.content === '';
+		const blockBHastEmptyContent = blockB?.attributes.content === '';
+		if ( blockAHasEmptyContent && ! blockBHastEmptyContent ) {
+			dispatch.removeBlock( clientIdA );
+
+			if ( canRestoreTextSelection ) {
+				dispatch.selectionChange(
+					blockB.clientId,
+					attributeKey,
```

Comment:
`attributeKey` comes from `getSelectionStart()`. I don't think that's guaranteed to be block B. Do we need to restore selection if we're just removing the previous block? Selection can just remain where it is without us having to do anything?

Reply:
Thanks for the tips! Last attempt I missed the `selectPrevious` args. Removed the `selectionChange` and rerun e2e testing to made sure it works properly